homeassistant/helpers/__init__.py

```diff
@@ -42,5 +41,4 @@ def extract_domain_configs(config: ConfigType, domain: str) -> Sequence[str]:
 
     Async friendly.
     """
-    pattern = re.compile(rf"^{domain}(| .+)$")
-    return [key for key in config if pattern.match(key)]
+    return [key for key in config if key.partition(" ")[0] == domain]
```

Comment:
How about this?

```suggestion
    parts = key.partition(" ")
    return [key for key in config if parts[0] == domain and parts[1]]
```

That would guard the case Marting is raising?

Reply:
You are right, sorry about that, adjusted.

Reply:
The suggested change is not good because not only does `domain ` fail, also `domain` fails:
```
>>> "domain".partition(" ")
('domain', '', '')
```

Reply:
You are absolutely right 👍